src/pyproject.toml

```diff
@@ -18,6 +18,7 @@ pinecone-client = "^2.2.2"
 beautifulsoup4 = "4.12.2"
 markdown = ">=3.4.4,<3.5.0"
 loguru = ">=0.7.0,<0.8.0"
+boto3 = "^1.28.62"
```

Comment:
Hmmm, can we somehow make this dependency optional? I think we don't need this if the code is running locally, right?